src/Stratis.Bitcoin.Features.BlockStore/BlockStoreFeature.cs

```diff
@@ -84,7 +87,15 @@ private void AddInlineStats(StringBuilder log)
 
         public override Task InitializeAsync()
         {
-            this.connectionManager.Parameters.TemplateBehaviors.Add(new BlockStoreBehavior(this.chain, this.chainState, this.loggerFactory, this.consensusManager));
+            // Use ProvenHeadersBlockStoreBehavior for PoS Networks
+            if (this.network.Consensus.IsProofOfStake)
```

Comment:
I don't like to use this flag but for now ok.

Reply:
https://github.com/stratisproject/StratisBitcoinFullNode/pull/2615#discussion_r228702266

ehm... conflicts? :D
what's the best way and why?